src/adapters/index.ts

```diff
@@ -33,6 +33,7 @@ export interface Adapter {
     get<T>(ctor: ModelCtor<T>, value: any, opts?: GetOptions): Promise<T[]>;
     getOne<T>(ctor: ModelCtor<T>, value: any, opts?: GetOptions): Promise<T>;
     join<M>(model: M, relationshipKey: string, opts?: JoinOptions): Promise<M>;
+    count<M>(model: M, predicate: (model: M) => boolean): Promise<number>;
```

Comment:
I think the predicate here is a bit rethink specific which i'd like to avoid, also this typing isn't really correct for rethink anyway